datastore/elastic_search/query.py

```diff
@@ -47,6 +58,264 @@ def dictionary_query(connection, index_name, doc_type, entity_name, **kwargs):
+        ner_logger.debug('delete_entity_data_by_values: entity_name: {0} result {1}'.format(entity_name, str(result)))
+
+
+def add_entity_data(
+    connection, index_name, doc_type, entity_name, value_variant_records, **kwargs
+):
+    """
+    Save entity data in ES for the records
+
+    Args:
+        connection: Elasticsearch client object
+        index_name: The name of the index
+        doc_type: The type of the documents that will be indexed
+        entity_name (str): name of the entity for which the data is to be created
+        value_variant_records (list): List of records to be created in ES
+    Returns:
+        None
+    """
+    str_query = []
+    for record in value_variant_records:
+        query_dict = {
+            '_index': index_name,
+            '_op_type': 'index',
+            '_type': doc_type,
+            'dict_type': DICTIONARY_DATA_VARIANTS,
+            'entity_data': entity_name,
+            'language_script': record.get('language_script'),
+            'value': record.get('word'),
+            'variants': record.get('variants'),
+        }
+        str_query.append(query_dict)
+        if len(str_query) > constants.ELASTICSEARCH_BULK_HELPER_MESSAGE_SIZE:
+            helpers.bulk(connection, str_query, stats_only=True, **kwargs)
+            str_query = []
+
+    if str_query:
+        helpers.bulk(connection, str_query, stats_only=True, **kwargs)
+
+
+def get_entity_unique_values(
+        connection, index_name, doc_type, entity_name,
+        value_search_term=None, variant_search_term=None, empty_variants_only=False,
+        **kwargs):
+    """
+    Search for values in entity with filters
+
+    Args:
+        connection: Elasticsearch client object
+        index_name: The name of the index
+        doc_type: The type of the documents that will be indexed
+        entity_name (str): name of the entity for which the data is to be fetched
+        value_search_term (str): Filter values with the specific search term
+        variant_search_term (str): Filter variants with the specific search term
+        empty_variants_only (bool): Search for values with empty variants only
+    Returns:
+        values (list): List of values which match the filters and search criteria
+    """
+    data = {
+        "sort": [
+            {
+                "value.keyword": {
+                    "order": "asc"
+                }
+            }
+        ],
+        "query": {
+            "bool": {
+                "must": [
+                    {
+                        "match": {
+                            "entity_data": entity_name
+                        }
+                    }
+                ],
+                "minimum_should_match": 0,
+                "should": []
+            }
+        },
+        "aggs": {
+            "unique_values": {
+                "terms": {
+                    "field": "value.keyword",
+                    "size": constants.ELASTICSEARCH_SEARCH_SIZE
```

Comment:
this limit might be crossed for certain entities

Reply:
handled by scroll